### DISCUSSION TITLE
FE PorousFlow applying mechanical stress and units

### DISCUSSION THREAD

**Discussion by srinath-chakravarthy** [idaholab/moose#21313] [Q&A Modules: General] [answered]
Hello All, 
I am trying to a porousflow simulation of a 1phase fluid, infiltrating a porous block that is initially unsaturated. The boundary conditions are PorousFlowSink at one boundary. This simulation works fine, but if i wanted the block to have a constant mechanical stress applied at the boundary, how would i go about this ? Input file is below the units description. 

Second question, the exact same input file with the following units does not converge. 
```
# -------------- Base Units ------------------------- #
# Length              = 1e-6 m -> mum
# Charge              = 1e-9 C -> nC
# Amt of substance    = 1e-12 mol -> pmol
# Temperature         = K
# Mass                = kg
# Time                = s
# # -------------- Derived Quantities -------------------#
# Force                = 1e-6 N -> muN
# Stress/Pressure     = 1e6 N/m^2 => MPa
# Energy              = 1e-12 J -> pJ
#
# Concentration       = 1e6 mol/m^3  -> pmol/mum^3
# Concentration Flux  = mol/m^2/s  -> pmol/mum^2/s
# Diffusivity         = 1e-12 m^2/s -> mum^2/s
# Chemical Potential  = J/mol -> pJ/pmol
# Molar Volume        = 1e-6 m^3/mol -> mum^3/pmol
#
#
# Current             = 1e-9 A => nA => nC/s
# Current density     = 1e3 A/m^2 => nA/mum^2
# Resistance          = 1e6 ohm => kgm^2/s^3/A^2 => kg mu^2 s^-3 nA^-2 => M-ohm
# Resistivity         = ohm-m
# Electric Field      = 1e3 V/m
# Conductivity        = S/m
# Charge transfer res = 1e-6 ohm m^2
#
# Thermal Conduc      = 1e-6 W/m/K
#
# # ----------------------------------------------------
# Gas Constant        = 8.314 J/mol/K
# Faradays constant   = 96.4853329 nC/pmol
# # ----------------------------------------------------
# Porous flow quantities
# Permeability        = 1e-12 m^2 -> 1 um^2
# All fractions remain fractions
# PorousFlowSink      = actual units => kg m^-2 s^-1 => kg um^2 s^-1 => 1e-12
# So current density 1A/m^2 => 66.143144e-9 kg/m^2s => 66.134144e-21 kg/um^2s
# Dynamic viscosity => kg/m^1/s => kg/um/s => 1e-6 kg/m/s
# Density           = kg/m^3 => kg/um^3 = 1e-18 kg/m^3

`[Mesh]
  [mesh]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 60
    xmin = 0.0
    xmax = 100e-6
    bias_x = 1
    ny = 30
    ymin = 0
    ymax = 30e-6
  []

  [injection_area]
    type = ParsedGenerateSideset
    combinatorial_geometry = 'y<0.0001e-6'
    included_subdomain_ids = 1
    new_sideset_name = 'injection_area'
    input = 'mesh'
  []


[]

[Problem]
  coord_type = RZ
  type = ReferenceResidualProblem
  extra_tag_vectors = 'ref'
  reference_vector = 'ref'
  acceptable_iterations = 2
  group_variables = 'disp_r disp_z temperature porepressure'
[]

[GlobalParams]
  displacements = 'disp_r disp_z'
  PorousFlowDictator = dictator
  biot_coefficient = 1.0
[]

[Variables]
  [porepressure]
  []
  [temperature]
    initial_condition = 293
  []
  [disp_r]
  []
  [disp_z]
  []
[]

[PorousFlowUnsaturated]
  porepressure = porepressure

  temperature = temperature
  coupling_type = ThermoHydroMechanical
  gravity = '0 0 0'
  fp = the_simple_fluid
  eigenstrain_names = thermal_contribution
  use_displaced_mesh = false
  multiply_by_density = true
  relative_permeability_exponent = 3
  relative_permeability_type = Corey
  residual_saturation = 0.001
  van_genuchten_alpha = 1e-6
  van_genuchten_m = 0.6
[]

[BCs]
  # [cavity_pressure]
  #   type = DirichletBC
  #   variable = porepressure
  #   value = 1e5
  #   boundary = top
  # []
  # [external_pressure]
  #   type = Pressure
  #   variable = disp_z
  #   use_displaced_mesh = false
  #   factor = -1e6
  #   displacements = 'disp_r disp_z'
  #   boundary = top
  #   extra_vector_tags = 'ref'
  # []
  [constant_injection_flux]
    type = PorousFlowSink
    variable = porepressure
    flux_function = "if (t <= 5000,if (t <10, -66.41344e-9*t,-66.41344e-9*10), 66.41344e-9)"
    boundary = bottom
    fluid_phase = 0
    use_relperm = true
    extra_vector_tags = 'ref'
    save_in = nodal_outflow
  []
  [constant_injection_temperature]
    type = DirichletBC
    variable = temperature
    value = 293
    boundary = bottom
  []
  [roller_tmax]
    type = DirichletBC
    variable = disp_r
    value = 0
    boundary = 'left'
  []
  [roller_top_bottom]
    type = DirichletBC
    variable = disp_z
    value = 0
    boundary = 'top'
  []
[]

[Postprocessors]

  [li_mass]
    type = PorousFlowFluidMass
    PorousFlowDictator = dictator
    fluid_component = 0
  []
  [porepressure_bottom]
    type = SideAverageValue
    variable = porepressure
    boundary = bottom
  []
  [stress_zz_bottom]
    type = SideAverageValue
    variable = stress_yy
    boundary = bottom
  []
  [saturation]
    type = SideAverageValue
    variable = saturation0
    boundary = bottom
  []
  [total_current]
    type = PorousFlowFluidMass
    PorousFlowDictator = dictator
    fluid_component = 0
  []
  [avg_permeability]
    type = ElementAverageValue
    variable = permeability
  []
[]

[AuxVariables]
  [nodal_outflow]
  []
  [porosity]
    family = MONOMIAL
    order = CONSTANT
  []
  [permeability]
    family = MONOMIAL
    order = CONSTANT
  []
  [stress_xx]
    family = MONOMIAL
    order = CONSTANT
  []
  [stress_yy]
    family = MONOMIAL
    order = CONSTANT
  []
  [hydrostatic_stress]
    family = MONOMIAL
    order = CONSTANT
  []
  [vonmises_stress]
    family = MONOMIAL
    order = CONSTANT
  []
[]

[AuxKernels]
  [porosity]
    type = PorousFlowPropertyAux
    property = porosity
    variable = porosity
  []
  [permeability]
    type = PorousFlowPropertyAux
    property = permeability
    column = 1
    row = 1
    variable = permeability
  []
  [stress_xx]
    type = RankTwoAux
    rank_two_tensor = stress
    variable = stress_xx
    index_i = 0
    index_j = 0
  []
  [stress_yy]
    type = RankTwoAux
    rank_two_tensor = stress
    variable = stress_yy
    index_i = 1
    index_j = 1
  []
  [hydrostatic_stress]
    type = RankTwoScalarAux
    rank_two_tensor = stress
    scalar_type = Hydrostatic
    variable = hydrostatic_stress
  []
  [vonmises_stress]
    type = RankTwoScalarAux
    rank_two_tensor = stress
    scalar_type = VonMisesStress
    variable = vonmises_stress
  []

[]

[Modules]
  [FluidProperties]
    [the_simple_fluid]
      type = SimpleFluidProperties
      bulk_modulus = 11E9
      viscosity = 1.0e4
      density0 = 534
      thermal_expansion = 0.0000118
      cp = 4194
      cv = 4186
      porepressure_coefficient = 0
    []
  []
[]

[Materials]
  [porosity]
    type = PorousFlowPorosityConst
    porosity = 0.2
    chemical = false
  []

  [biot_modulus]
    type = PorousFlowConstantBiotModulus
    solid_bulk_compliance = 2E-9
    fluid_bulk_modulus = 11E9
  []
  [permeability]
    type = PorousFlowPermeabilityKozenyCarman
    k0 = 1e-12
    k_anisotropy = '1E-25 0 0   0 1 0   0 0 1E-25'
    f = 0.01
    d = 50e-9
    poroperm_function = kozeny_carman_fd2
    m = 3
    n = 2
  []

  [thermal_expansion]
    type = PorousFlowConstantThermalExpansionCoefficient
    drained_coefficient = 0.00002
    fluid_coefficient = 0.00002
  []
  [rock_internal_energy]
    type = PorousFlowMatrixInternalEnergy
    density = 2500.0
    specific_heat_capacity = 1200.0
  []
  [thermal_conductivity]
    type = PorousFlowThermalConductivityIdeal
    dry_thermal_conductivity = '10 0 0  0 10 0  0 0 10'
  []

  [elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 100E9
    poissons_ratio = 0.25
  []
  [strain]
    type = ComputeSmallStrain
    eigenstrain_names = thermal_contribution
  []
  [thermal_contribution]
    type = ComputeThermalExpansionEigenstrain
    temperature = temperature
    thermal_expansion_coeff = 1e-19 # this is the linear thermal expansion coefficient
    eigenstrain_name = thermal_contribution
    stress_free_temperature = 293
  []
  [stress]
    type = ComputeLinearElasticStress
  []
[]

[Preconditioning]
  active = preferred_but_might_not_be_installed
  [basic]
    type = SMP
    full = true
    petsc_options = '-ksp_diagonal_scale -ksp_diagonal_scale_fix'
    petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap'
    petsc_options_value = ' asm      lu           NONZERO                   2'
  []
  [preferred_but_might_not_be_installed]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Transient
  automatic_scaling = true
  # compute_scaling_once = false
  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package -snes_force_iteration'
  petsc_options_value = ' lu       mumps 1'

  solve_type = Newton
  # end_time = 36000
  dt = 0.1
  nl_max_its = 100
  nl_abs_tol = 1E-8
  nl_rel_tol = 1E-3
  resid_vs_jac_scaling_param = 0.5
  dtmax = 25
  end_time = 10000
  # num_steps = 10
  [TimeStepper]
    type = IterationAdaptiveDT
    dt = 1
    optimal_iterations = 100
    cutback_factor = 0.5
    growth_factor = 1.1
  []

[]

[Outputs]
  exodus = true
  csv = true
  sync_times = '500'
  sync_only = false
  file_base = xxx2
[]
```

**Comment by WilkAndy** [comment]
- A constant mechanical stress... This depends on what exactly you want to achieve.  There are some examples in the "Barometric and ocean loading" section of https://mooseframework.inl.gov/modules/porous_flow/tidal.html .    Note that all stresses in porousflow are effective stresses, *except* the boundary conditions, which are in terms of the total stress.  Physically, the total stresses are brought about by something pushing on the rock skeleton.  There are two distinct cases.  (1) If your constant mechanical stress is due to some object pushing on the rock skeleton (eg, a train sitting on the ground surface), then you should use a `FunctionNeumannBC` or a `PressureBC` applied to the appropriate `disp` variable.  On the other hand, (2), if your constant mechanical stress is due to *fluid* pushing on the rock skeleton (eg, a borehole pressure), then you should use the aforementioned `FunctionNeumannBC` or `PressureBC`, along with an appropriate boundary condition for the fluid porepressure, such as `DirichletBC` on the `porepressure` variable.
- Using different units is sometimes tricky.  Please see the essay at "An essay on Pascals, kilograms and densities" of https://mooseframework.inl.gov/source/materials/PorousFlowSingleComponentFluid.html .  Once you're sure of the numerical quantities in your input file, any nonconvergence must come from inappropriately scaled variables or tolerances.   There is an old page on scaling here https://mooseframework.inl.gov/modules/porous_flow/convergence.html but now i find
```
[Debug]
  show_var_residual_norms = True
[]
```
to be a quicker way of determining roughly the correct scaling.   That page will also tell you reasonable tolerances.  Eg, if you use mm rather than m, probably your residuals will increase by 1E9, so absolute tolerances have to be modified.  By the way, i'm not a super fan of autoscaling, because often i set absolute tolerances in my models.

**Comment by srinath-chakravarthy** [comment]
Regarding, the stress part, assuming I have a train sitting on the rock,
then the pressure bc that is commented out in the input file should work as
a total stress applied on the skeleton, but the model when run with those
lines uncommented, seems to produce stresses I cannot understand or
interpret. Can you help with t hi is ?

Regarding units, thanks for your suggestion. Just to be clear I scaled the
problem in tutorial 8 of the porous flow middle module and replaced them
with the scaling I have detailed and that seemed to work just fine.

Regarding scaling, I have a much more complicated problem and several
blocks with different physics I need to solve,nor which auto scaling with
reference residual problem eliminates some of the need for abs tolerances
and I can deal with relative ones. Hence the auto scaling..the input file
is the simplest picture of porous flow that I could present

**Reply by WilkAndy** [reply]
Remember that the output stresses (in paraview or postprocessors) are effective stresses, not total stresses.  Does that help?

**Comment by srinath-chakravarthy** [comment]
So just the sun of the output value and pore pressure should do the trick
??

**Reply by WilkAndy** [reply]
effective = total + biot_coefficient * porepressure (see https://mooseframework.inl.gov/modules/porous_flow/governing_equations.html ).   So you want effective - biot_coefficient * porepressure   (since your biot_coefficient = 1, you need effective - porepressure).

**Comment by srinath-chakravarthy** [comment]
Andy,
More questions regarding the simple model i have presented.
1) Let us assume an initially empty reservoir (porous skeleton) at
atmospheric pressure. I want to start pumping fluid into this skeleton. I
am using a porousflowsink to do this. What should be the boundary
conditions on the end away from the pumping ? I want to model how long it
will take to fill this reservoir. So shpuld the initial condition be
porepressure = -1e5 (patm) ?
2) For the same situation, i would like to apply a mechanical stress on the
end away from the pumping. So in effect the skeleton will exert a pressure
on the pump. In the thought experiment, the filling of the reservoir should
not be affected by this external mechanical stress, since the only thing
that governs the flow is the pressure gradient. Presumably the
initial condition here will be porepressure = -\sigma will be the boundary
condition at the pumping site.

Now the question on how to apply this initial condition. I read the
documentation and there documentation on applying postprocessor values to
the porepressure. What post-processor value am i going to be choosing here
?

Cheers
Srinath

**Reply by WilkAndy** [reply]
- Fluid boundary conditions at the "far end"....  My initial thought was to use PorousFlowOutflowBC https://mooseframework.inl.gov/source/bcs/PorousFlowOutflowBC.html .   However, that is really designed to model an infinite reservoir (see the documentation for a discussion).  I suspect you're got a finite reservoir, given that you're applying a mechanical stress at the "far end".  In that case, you should probably use a PorousFlowSink, as discussed in https://mooseframework.inl.gov/modules/porous_flow/boundaries.html .
- Initially unsaturated configuration could be p=-1E5 Pascals, as you've indicated.  But that depends on your capillary curve.  "initially unsaturated" can mean different things to different people.  Viz, there is usually residual water that exists in natural systems - perhaps 5% water, perhaps even 20% water, depending on the capillarity and the history of the material (eg, if it has been dried using high temperature and then kept in a vacuum, it'd have tiny saturation).
- Assuming the mechanical stress and strain don't impact porosity or permeability (this is usually a reasonable assumption but you may wish to relax it) then mechanical stress won't impact fluid flow.  But if that's the case then you might like to consider whether there's any point building a model with both displacements and fluid: why not build two separate models (one for the fluid flow, one for the mechanics).
- The boundary condition at the pumping site will be compressive normal total stress = pumping porepressure, and porepressure = pumping pressure.
- Sorry, i'm not 100% sure what the postprocessor is for.  Are you injecting at a known rate, but don't know the porepressure?  If so, then study the "Boundary conditions" section of https://mooseframework.inl.gov/modules/porous_flow/thm_example.html

**Comment by srinath-chakravarthy** [comment]
Andy,
Thanks a lot. Let me explain a little more about the problem.
We have a micro scale porous media attached to a permeabeable membrane,
below which there is a comical chemical pump, pumping constant rate of
fluid thru the membrane into the porous skeleton. Above certain values of
the rate we observe separation between the membrane and the skeleton. One
solution to this problem is applying a mechanical pressure to prevent the
delamination of the membrane.  While absolute values of the stress/pressure
does not affect the fluid flow, absolute values do affect the delamination.
I have already implemented a cohesive zone type model for the mass flux to
stop if the interface separation is greater than a certain value. This is
why I need  displacement coupling to the fluid flow.

Capillary curve : completely understood about what 0 saturation is, and I
was just giving you an example of what the choice of initial conditions
will be.

Pumping end BCS, are currently fixed displacementn in direction normal to
the flow. The pore pressure will presumably come from a post processor. I
am however confused as to how one can apply a Neumann and a Dirichlet bc
for the same variable at the same boundary. In this instance we have a
porousflowsink and a pressure bc. I see the physical need for it, but
cannot quite understand the implementation.

Thanks for your help to a newbie on porous flow mechanics.

Cheers
Srinath

**Comment by srinath-chakravarthy** [comment]
Andy, 
The results don't seem to be make any sense at all. Please see input file below. There is no effect of the applied mechanical stress on the effective or total stress in the system or the absolute pore-pressure. I am thoroughly confused. Perhaps this has to do with unsaturated flow. 
`
[Mesh]
  [mesh]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 60
    xmin = 0.0
    xmax = 100e-6
    bias_x = 1
    ny = 30
    ymin = 0
    ymax = 30e-6
  []

  [injection_area]
    type = ParsedGenerateSideset
    combinatorial_geometry = 'y<0.0001e-6'
    included_subdomain_ids = 1
    new_sideset_name = 'injection_area'
    input = 'mesh'
  []


[]

[Problem]
  coord_type = RZ
  type = ReferenceResidualProblem
  extra_tag_vectors = 'ref'
  reference_vector = 'ref'
  acceptable_iterations = 2
  group_variables = 'disp_r disp_z temperature porepressure'
[]

[GlobalParams]
  displacements = 'disp_r disp_z'
  PorousFlowDictator = dictator
  biot_coefficient = 1.0
[]

[Variables]
  [porepressure]
  []
  [temperature]
    initial_condition = 293
  []
  [disp_r]
  []
  [disp_z]
  []
[]

[PorousFlowUnsaturated]
  porepressure = porepressure

  temperature = temperature
  coupling_type = ThermoHydroMechanical
  gravity = '0 0 0'
  fp = the_simple_fluid
  eigenstrain_names = thermal_contribution
  use_displaced_mesh = false
  multiply_by_density = true
  relative_permeability_exponent = 3
  relative_permeability_type = Corey
  residual_saturation = 0.001
  van_genuchten_alpha = 5e-5
  van_genuchten_m = 0.6
[]

[BCs]

  [external_pressure]
    type = Pressure
    variable = disp_z
    use_displaced_mesh = false
    boundary = top
    factor = 1e6
  []

  [pumping_site_pp]
    type = Pressure
    variable = disp_z
    postprocessor = pumping_pressure
    boundary = bottom
    use_displaced_mesh = false
    extra_vector_tags = 'ref'
  []


  [constant_injection_flux]
    type = PorousFlowSink
    variable = porepressure
    flux_function = "if (t <= 5000,if (t <10, -66.41344e-9*t,-66.41344e-9*10), 66.41344e-9)"
    boundary = bottom
    fluid_phase = 0
    use_relperm = true
    extra_vector_tags = 'ref'
    save_in = nodal_outflow
  []
  [constant_injection_temperature]
    type = DirichletBC
    variable = temperature
    value = 293
    boundary = bottom
  []
  [roller_tmax]
    type = DirichletBC
    variable = disp_r
    value = 0
    boundary = 'left right'
  []
  [roller_top_bottom]
    type = DirichletBC
    variable = disp_z
    value = 0
    boundary = 'top'
  []
[]

[ICs]
  [pp]
    type = ConstantIC
    value = -1e5
    variable = porepressure
  []
[]
[Postprocessors]
  [pumping_pressure]
    type =  SideAverageValue
    variable = porepressure
    boundary = bottom
    execute_on = TIMESTEP_BEGIN
  []

  [mass_flow]
    type = PorousFlowFluidMass
    PorousFlowDictator = dictator
    fluid_component = 0
  []
  [porepressure_bottom]
    type = SideAverageValue
    variable = porepressure
    boundary = bottom
  []
  [effective_stress_zz_bottom]
    type = SideAverageValue
    variable = stress_yy
    boundary = bottom
  []
  [total_stress_bottom]
    type = ParsedPostprocessor
    function = 'effective_stress_zz_bottom - porepressure_bottom'
    pp_names = 'effective_stress_zz_bottom porepressure_bottom'
  []
  [saturation]
    type = SideAverageValue
    variable = saturation0
    boundary = bottom
  []

[]

[AuxVariables]
  [nodal_outflow]
  []
  [porosity]
    family = MONOMIAL
    order = CONSTANT
  []
  [permeability]
    family = MONOMIAL
    order = CONSTANT
  []
  [stress_xx]
    family = MONOMIAL
    order = CONSTANT
  []
  [stress_yy]
    family = MONOMIAL
    order = CONSTANT
  []
  [hydrostatic_stress]
    family = MONOMIAL
    order = CONSTANT
  []
  [vonmises_stress]
    family = MONOMIAL
    order = CONSTANT
  []
[]

[AuxKernels]
  [porosity]
    type = PorousFlowPropertyAux
    property = porosity
    variable = porosity
  []
  [permeability]
    type = PorousFlowPropertyAux
    property = permeability
    column = 1
    row = 1
    variable = permeability
  []
  [stress_xx]
    type = RankTwoAux
    rank_two_tensor = stress
    variable = stress_xx
    index_i = 0
    index_j = 0
  []
  [stress_yy]
    type = RankTwoAux
    rank_two_tensor = stress
    variable = stress_yy
    index_i = 1
    index_j = 1
  []
  [hydrostatic_stress]
    type = RankTwoScalarAux
    rank_two_tensor = stress
    scalar_type = Hydrostatic
    variable = hydrostatic_stress
  []
  [vonmises_stress]
    type = RankTwoScalarAux
    rank_two_tensor = stress
    scalar_type = VonMisesStress
    variable = vonmises_stress
  []

[]

[Modules]
  [FluidProperties]
    [the_simple_fluid]
      type = SimpleFluidProperties
      bulk_modulus = 11E9
      viscosity = 1.0e1
      density0 = 534
      thermal_expansion = 0.0000118
      cp = 4194
      cv = 4186
      porepressure_coefficient = 0
    []
  []
[]

[Materials]
  [porosity]
    type = PorousFlowPorosityConst
    porosity = 0.2
    chemical = false
  []

  [biot_modulus]
    type = PorousFlowConstantBiotModulus
    solid_bulk_compliance = 2E-9
    fluid_bulk_modulus = 11E9
  []
  [permeability]
    type = PorousFlowPermeabilityKozenyCarman
    k0 = 1e-12
    k_anisotropy = '1E-25 0 0   0 1 0   0 0 1E-25'
    f = 0.01
    d = 50e-9
    poroperm_function = kozeny_carman_fd2
    m = 3
    n = 2
  []

  [thermal_expansion]
    type = PorousFlowConstantThermalExpansionCoefficient
    drained_coefficient = 0.00002
    fluid_coefficient = 0.00002
  []
  [rock_internal_energy]
    type = PorousFlowMatrixInternalEnergy
    density = 2500.0
    specific_heat_capacity = 1200.0
  []
  [thermal_conductivity]
    type = PorousFlowThermalConductivityIdeal
    dry_thermal_conductivity = '10 0 0  0 10 0  0 0 10'
  []

  [elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 100E9
    poissons_ratio = 0.25
  []
  [strain]
    type = ComputeSmallStrain
    eigenstrain_names = thermal_contribution
  []
  [thermal_contribution]
    type = ComputeThermalExpansionEigenstrain
    temperature = temperature
    thermal_expansion_coeff = 1e-19 # this is the linear thermal expansion coefficient
    eigenstrain_name = thermal_contribution
    stress_free_temperature = 293
  []
  [stress]
    type = ComputeLinearElasticStress
  []
[]

[Preconditioning]
  active = preferred_but_might_not_be_installed
  [basic]
    type = SMP
    full = true
    petsc_options = '-ksp_diagonal_scale -ksp_diagonal_scale_fix'
    petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap'
    petsc_options_value = ' asm      lu           NONZERO                   2'
  []
  [preferred_but_might_not_be_installed]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Transient
  automatic_scaling = false
  # compute_scaling_once = false
  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package -snes_force_iteration'
  petsc_options_value = ' lu       mumps 1'

  solve_type = Newton
  # end_time = 36000
  dt = 0.1
  nl_max_its = 100
  nl_abs_tol = 1E-8
  nl_rel_tol = 1E-3
  resid_vs_jac_scaling_param = 0.5
  dtmax = 25
  end_time = 500
  # num_steps = 10
  [TimeStepper]
    type = IterationAdaptiveDT
    dt = 1
    optimal_iterations = 100
    cutback_factor = 0.5
    growth_factor = 1.1
  []

[]

[Outputs]
  exodus = true
  csv = true
  sync_times = '500'
  sync_only = false
  file_base = xxx2
[]
`

**Comment by WilkAndy** [comment, marked as answer]
Hi @srinath-chakravarthy ,

Just check your BCs.  For instance, it looks like you're applying a `Pressure` **and** a `DirichletBC` on `top` to the Variable `disp_z`.  Perhaps a copy-and-paste error and your roller BCs aren't really what you wanted?   Does this help or am i way off?

a

**Comment by srinath-chakravarthy** [comment]
Sorry about the cut and paste error. So silly, I was barking up the wrong
tree... Thanks a lot. Will check with correct BCS in a little bit.

Cheers
Srinath

**Comment by srinath-chakravarthy** [comment]
Perfect that works. Thanks
